dvc/utils/studio.py

```diff
@@ -111,3 +115,51 @@ def env_to_config(env: dict[str, Any]) -> dict[str, Any]:
     if DVC_STUDIO_URL in env:
         config["url"] = env[DVC_STUDIO_URL]
     return config
+
+
+def get_subrepo_relpath(repo: "Repo") -> str:
+    from dvc.fs import GitFileSystem
+
+    scm_root_dir = "/" if isinstance(repo.fs, GitFileSystem) else repo.scm.root_dir
+
+    relpath = as_posix(repo.fs.relpath(repo.root_dir, scm_root_dir))
+
+    return "" if relpath == "." else relpath
+
+
+def get_dvc_experiment_parent_data(
+    repo: "Repo", baseline_rev: Union[str, None]
+) -> Union[dict[str, Any], None]:
+    from scmrepo.exceptions import SCMError
+
+    from dvc.scm import NoSCM, get_n_commits
+
+    scm = repo.scm
+
+    try:
+        if (
+            not baseline_rev
+            or not scm
+            or isinstance(scm, NoSCM)
+            or not (commit := scm.resolve_commit(baseline_rev))
+            or not (
+                first_100_parent_shas := get_n_commits(scm, [baseline_rev], 101)[1:]
```

Comment:
Q: shy do we need this? (looks complicated, and not very obvious)

Reply:
See https://github.com/iterative/studio/issues/8848#issuecomment-1935151710

It is complicated and it is needed because of commits missing from Studio.

Reply:
Got it. A few thoughts / questions:

- should it be fine for us at least initially disregard it (and improve later?). Rely on the existence of the baseline commit in Studio? (if I understood the issue correctly, apologies if not)
- should it be optional (remove it from `if .. or not`) - so that's it's not a blocker for sending at least basic data?
- probably a place where a comment would not hurt :) (absolutely your call on this)

Reply:
> * should it be fine for us at least initially disregard it (and improve later?). Rely on the existence of the baseline commit in Studio? (if I understood the issue correctly, apologies if not)
> * should it be optional (remove it from `if .. or not`) - so that's it's not a blocker for sending at least basic data?

This data is additional to the `baseline_rev` not instead of. The extra data will only be used if the baseline cannot be found in Studio. Under those circumstances we will try to create a "fake" parent. Without the parent commits we will not know where to insert the "fake" parent and so will still end up with a detached commit. IMO there is therefore no point in sending the extra data without the parent commits.

This solutions is the best (and maybe only) available given the current constraints.

Reply:
Just to clarify and double check.`get_dvc_experiment_parent_data` would return `None` (no even baseline sent) if we can't get `  first_100_parent_shas` for some reason ... is it correct?

My question was - can there a be situation when getting just a single baseline works, but 100 fails.

Reply:
The baseline is duplicated in the payload sent for the `start` event. It is sent both inside and outside of the `dvc_experiment_parent_data` (smaller change). An example payload is shown in [this test](https://github.com/iterative/dvc/pull/10291/files#diff-8e0861755c45357459c7a213f4f011fafa6ed83733f4aa8e96b91fd6daecc815R15). I.e. 

```
        assert start_call.kwargs["json"] == {
            "type": "start",
            "repo_url": "STUDIO_REPO_URL",
            "baseline_sha": baseline_sha,
            "name": name,
            "params": {"params.yaml": {"foo": 1}},
            "client": "dvc",
            "dvc_experiment_parent_data": {
                "author": {
                    "email": "dvctester@example.com",
                    "name": "DVC Tester",
                },
                "date": M.any,
                "message": "init",
                "parent_shas": M.any,
                "title": "init",
                "sha": baseline_sha,
            },
        }
```

Reply:
> My question was - can there a be situation when getting just a single baseline works, but 100 fails.

Shouldn't happen but the `get_dvc_experiment_parent_data` is meant to be designed to not stop the `start` event being sent OR brick the experiment if the required data is not found.